Lib/test/_test_multiprocessing.py

```diff
@@ -3736,6 +3736,23 @@ def test_shared_memory_across_processes(self):
 
         sms.close()
 
+    @unittest.skipIf(WIN32, "test not feasible in Windows")
+    def test_shared_memory_SharedMemoryServer_ignores_sigint(self):
+        # bpo-36368: protect SharedMemoryManager server process from
+        # KeyboardInterrupt signals.
+        smm = multiprocessing.managers.SharedMemoryManager()
+        smm.start()
+
+        # make sure the manager works properly at the beginning
+        sl = smm.ShareableList(range(10))
+
+        # the manager's server should ignore KeyboardInterrupt signals, and
+        # maintain its connection with the current process
+        os.kill(smm._process.pid, signal.SIGINT)
```

Comment:
I think you should add a test sending `SIGINT` to the main process, showing that `SIGINT` is only affected in the manager. Something like
```
with self.assertRaises(KeyboardInterrupt):
    os.kill(os.getpid(), signal.SIGINT)
```